context/app/utils.py

```diff
@@ -116,3 +117,35 @@ def find_raw_dataset_ancestor(client, ancestor_ids):
             'is_component'
         ]
     )
+
+
+def get_epics_pyramid_entity(uuid):
+    """
+        Retrieves the entity for the base image-pyramid for EPIC segmentation masks
+        by searching through the descendants of the parent and looking for the latest
+        centrally processed dataset.
+
+        Parameters:
+        uuid (str): The unique identifier of the parent entity whose descendants will be searched.
+
+        Returns:
+            dict: The entity dictionary representing the most recent centrally processed
+            image-pyramid dataset, or None if no suitable dataset is found.
+
+    """
+    client = get_client()
+    entity = client.get_entity(uuid)
+    descendants = entity.get('descendant_ids')
+    max_modified_date = None
+    max_date_descendant = None
+    for descendant in descendants:
+        dec_entity = client.get_entity(descendant)
+        # TODO: Until we have better entity qualifier
+        if dec_entity.get('creation_action').lower() == 'central process' and len(
+                dec_entity.get('files')) > 0 and dec_entity.get('status').lower() != 'error':
```

Comment:
Historically we've lifted entities that are in `QA` or `Published`, should we continue that logic here?